listener.go

```diff
@@ -202,7 +207,7 @@ func (l *Listener) handleMsg(buf []byte, oob *ipv4.ControlMessage, _peer net.Add
 	// the default gateway handed out by DHCP is the .1 of whatever /24 subnet the client gets handed out.
 	// we actually don't care at all what the gw IP is, its really just to make the client's tcp/ip stack happy
 	if options.Gateway == nil {
-		gw := net.IPv4(pickedIP[0], pickedIP[1], pickedIP[2], 1)
+		gw := net.IPv4(pickedIP.IP[0], pickedIP.IP[1], pickedIP.IP[2], 1)
```

Comment:
this math would be incorrect if the subnet is != /24. maybe we should do the proper bit math to determine the correct gw IP by default?